components/suggest/src/db.rs

```diff
@@ -292,10 +292,10 @@ impl<'a> SuggestDao<'a> {
         Ok(())
     }
 
-    /// Inserts an icon for a suggestion into the database.
-    pub fn insert_icon(&mut self, icon_id: &str, data: &[u8]) -> Result<()> {
+    /// Inserts or replaces an icon for a suggestion into the database.
+    pub fn put_icon(&mut self, icon_id: &str, data: &[u8]) -> Result<()> {
         self.conn.execute(
-            "INSERT INTO icons(
+            "REPLACE INTO icons(
```

Comment:
Q: per SQLite's [doc](https://www.sqlite.org/lang_replace.html), `REPLACE` is a shorthand of `INSERT OR REPLACE` borrowed from MySQL, I'd recommend using the latter for clarity.

Also, icons are append-only on RemoteSettings, we'd never delete/modify icons once published to RS. So technically, we can also do `INSERT OR IGNORE`.

Reply:
Sounds good to me! I've been using `REPLACE` for a while, but let's use `INSERT OR REPLACE` in Suggest. I'll fix that here and in `put_meta`.

Good to know about icons being append-only, thanks! I think I'll keep it as `INSERT OR REPLACE` just in case, even though we don't expect updates, but it helps to know that this isn't an issue we'll expect to see in practice.